internal/services/firewall/firewall_policy_resource.go

```diff
@@ -925,6 +1020,15 @@ func resourceFirewallPolicySchema() map[string]*pluginsdk.Schema {
 			},
 		},
 
+		"auto_learn_private_ranges_mode": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(network.AutoLearnPrivateRangesModeEnabled),
+				string(network.AutoLearnPrivateRangesModeDisabled),
+			}, false),
+		},
+
```

Comment:
Would it make sense to make this property a boolean and call it `auto_learn_private_ranges_enabled` instead?

Reply:
that makes sense and more easy to use, but I wonder if there are other possible values in future

Reply:
Typically in cases where there is only enabled and disabled we tend to use a boolean anyway as it's not likely to have different values in future in this case.

Reply:
thanks, modified.